### PR TITLE
Hotfix #447

### PR DESCRIPTION
**Pull request by VicDeo**
if we want not to loose compatibility with 10.2

### CODE REVIEW

File: appinfo/register_command.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * @author Arthur Schiwon <blizzz@arthur-schiwon.de>
+ * @author Joas Schilling <coding@schilljs.com>
+ * @author Morris Jobke <hey@morrisjobke.de>
+ *
+ * @copyright Copyright (c) 2016, ownCloud GmbH.
```
Comment (DeepDiver1975) [inline appinfo/register_command.php]: 2019

File: appinfo/register_command.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * @author Arthur Schiwon <blizzz@arthur-schiwon.de>
```
Comment (DeepDiver1975) [inline appinfo/register_command.php]: please add correct author

**Comment by PVince81** [comment]
not sure if the hotfix is really needed, considering that the original issue is with the apps/apps_external constellation for which there is a workaround: https://github.com/owncloud/user_ldap/issues/447#issuecomment-521907751

unless we want to send this patch to users and not release / merge it ?

**Comment by VicDeo** [comment]
@DeepDiver1975 
>When creating the command from xml the exact same stuff is done as what has explicitly been coded now.
There is still something wrong if the code patch is not fixing the issue

true. 
but `register_command` is executed significantly later than `info.xml` is parsed. And when it is parsed wrong app.php is loaded. So the real fix is https://github.com/owncloud/core/pull/36054

and a hotfix for the app is partially revert  the changes from https://github.com/owncloud/user_ldap/commit/360af338bf93bb87a473144a53703ca1e2ff2bc2#diff-224bb875c5d3cfcd7621135069013242
This PR does exactly that. The same about authors - I just restored the original `register_command.php`, it had been not written by me.

**Comment by VicDeo** [comment]
@PVince81 I hadn't got any feedback so I decided to do both core fix and a hotfix that will provide a seamless upgrade experience and potentially reduce number of support cases.
Surely it's up to management to decide whether hotfix is needed or not. :)

**Comment by PVince81** [comment]
not wanting to merge this. People should update to 10.3 when it's out.

if needed, this PR can be sent as patch.

@pmaier1 @cdamken FYI

**Comment by pmaier1** [comment]
Please make sure that there will be no issues in upgrading LDAP to 0.14.0, also on server versions below 10.3.

**Comment by PVince81** [comment]
@VicDeo please adjust based on comments

maybe also add a PHP comment somewhere as a note that the command is not in info.xml as a workaround, for the future.

we'll need to keep in mind to revert this again for the next LDAP app version 0.15.0 or later

**Comment by VicDeo** [comment]
@PVince81 updated, added fixme

**Pull request merged**